plugins/explore/src/plugin.ts

```diff
@@ -16,21 +16,43 @@
 
 import { exploreToolsConfigRef } from '@backstage/plugin-explore-react';
 import { catalogEntityRouteRef, exploreRouteRef } from './routes';
-import { exampleTools } from './util/examples';
-import { createApiFactory, createPlugin } from '@backstage/core-plugin-api';
+import {
+  createApiFactory,
+  createPlugin,
+  discoveryApiRef,
+  fetchApiRef,
+} from '@backstage/core-plugin-api';
+import { ExploreClient, exploreApiRef } from './api';
 
 /** @public */
 export const explorePlugin = createPlugin({
   id: 'explore',
   apis: [
-    // Register a default for exploreToolsConfigRef, you may want to override
-    // the API locally in your app.
+    createApiFactory({
+      api: exploreApiRef,
+      deps: {
+        discoveryApi: discoveryApiRef,
+        fetchApi: fetchApiRef,
+      },
+      factory: ({ discoveryApi, fetchApi }) =>
+        new ExploreClient({ discoveryApi, fetchApi }),
+    }),
+    /**
+     * @deprecated Use ExploreApi from `@backstage/plugin-explore` instead
```

Comment:
There's a bit of a breaking change with the `explore-backend` needing to be installed as well right? I wonder if it makes sense to figure out some migration path there as well or not. Might be enough to call it out clearly in the changeset.

Reply:
Yep, 100%, potentially a breaking change. My thought process here was that the odds this not being overriden for an actual production instance would be low since it's hardcoded to the example app. However, this could be a bit misguided. I do want to make this a non breaking change so I'll think on this a bit more.

Reply:
I've thought about this and I can't come up with a nice way to make this 100% backwards compatible. My thinking is to just make it a breaking change in the changeset?

I've considered a few combinations, but ideally the `ToolExploreContent` component needs to eventually use the new `ExplorAPI/Client`. I'd also like to somehow warn about this, but not sure we have any patterns to do this on the frontend like we do on the backend with `logger.warn`?

Reply:
Yep, was thinking about some ways to detect whether there's an override in place or not, but in the end I think you're right in that it's best to just make this a breaking change in the way that you have it in place already. With a future removal of the tools config API.